v-next/hardhat-chai-matchers/src/internal/utils/deep-equal.ts

```diff
@@ -0,0 +1,17 @@
+// eslint-disable-next-line no-restricted-imports -- we want to cases like: 1n === 1
+import assert from "node:assert";
+
+/**
```

Comment:
Is this solution ok? Do we want to modify it, or keep the [npm package deep-eql](https://www.npmjs.com/package/deep-eql) that is build to work with `chai`?